### PR TITLE
Add message box with keybindings bound to '?' key.

### PR DESCRIPTION
**Pull request by zakkak**
messagebox

### CODE REVIEW

**Comment by dannyedel** [comment]
Thanks, this is a really good idea and its very nice to do it in Pull-Request form : )

I'll be reading it through now, and either merge or give comments at the points in code.
One thing I see though: There's a lot of whitespace changes, which makes it a bit hard to see the actual changes.

While that is obviously my fault for commiting such.. uhm.. things (trailing whitespace, blank lines at eof etc...) I'm a bit disappointed that kdevelop/kate did not prevent me from doing that in the first place. (Or I have not found the correct configuration)

I've read up and the [github blog](https://github.com/blog/967-github-secrets) says that by [appending w=1](https://github.com/dannyedel/dspdfviewer/pull/29/files?w=1) we can filter whitespace changes while viewing.

Side question: Do you know a good tool/editor to quickly walk through all those files, fix whitespace and commit? Maybe worth it's own pullrequest "fix whitespace"...
I'll get on to reading the actual code now.

File: pdfviewerwindow.cpp [hunk 1]
```diff
@@ -353,6 +357,28 @@ void PDFViewerWindow::updateWallClock(const QTime& wallClock)
...
 
+void PDFViewerWindow::keybindingsPopup()
+{
+  QMessageBox *popup   = new QMessageBox();
+  QString msg;
+  msg.append("<table>\n");
+  msg.append("<tr><th width=200 align=left>Key</th><th width=400 align=left>Action</th></tr>\n");
+  /* I skip some navigation bindings (they have to many alternatives) */
+  msg.append("<tr><td>N or Left/Down arrow</td><td>Next slide</td></tr>\n");
+  msg.append("<tr><td>P or RIght/Up arrow</td><td>Previous slide</td></tr>\n");
```
Comment (dannyedel) [inline pdfviewerwindow.cpp]: Typo: RIght

File: pdfviewerwindow.cpp [hunk 2]
```diff
@@ -353,6 +357,28 @@ void PDFViewerWindow::updateWallClock(const QTime& wallClock)
...
+  msg.append("<tr><td>B or .</td><td>Blank/Unblank audience screen</td></tr>\n");
+  msg.append("<tr><td>G</td><td>Go to specific slide</td></tr>\n");
+  msg.append("<tr><td>H or Home</td><td>Go to first page and reset counters</td></tr>\n");
+  msg.append("<tr><td>Q or Esc</td><td>Quit</td></tr>\n");
+  msg.append("<tr><td>S or F12</td><td>Switch primary and secondary screens</td></tr>\n");
+  msg.append("<tr><td>T</td><td>Toggle between notes and slides in the secondary screen</td></tr>\n");
+  msg.append("<tr><td>?</td><td>Show that help box</td></tr>\n");
```
Comment (dannyedel) [inline pdfviewerwindow.cpp]: Maybe "this" help box would be better since the user is already on it

**Comment by dannyedel** [comment]
Found two wording issues, otherwise cool!

**Comment by zakkak** [comment]
Sorry about the whitespace changes, it is my editor setup that removes trailing spaces.

Another "annoying" IMO thing is that indentation is not consistent.  The code is indented using spaces and whenever a group of spaces can be reduced to a tab it gets replaced.  I suggest either using solely tabs or spaces.

I am actually a fan of _indenting with tabs and aligning with spaces_ there is a nice example at the [emacswiki](http://www.emacswiki.org/emacs/SmartTabs).

smarttabs

**Comment by zakkak** [comment]
Regarding your side question. Both vim end emacs have options to remove trailing spaces, I do not remember if kate was capable of doing the same, if not there is probably some plugin.

Another approach is the use of tools like `uncrustify` that apart from removing whitespace allows for a consistent code style.  Unfortunately though, it is a pain to configure to your needs and is not available in every distribution (you most likely need to compile from source).

**Comment by dannyedel** [comment]
Thanks for this function! Merge on the way. Whitespace-issue redirected to #30

**Pull request merged**